### DISCUSSION TITLE
Mixing of CPU Detectors and Coral TPU?

### DISCUSSION THREAD

**Discussion by ajguerre1** [blakeblackshear/frigate#3322] [Ask A Question] [answered]
Hi,

Can you mix CPU detectors with Coral TPUs? The documentation is not 100% clear if that is possible or even stable.

**Comment by NickM-27** [comment]
Is there a reason you want to do this? 

Only case this could be needed is if the coral can't handle the number of cameras you have. 

The cpu won't help the coral be faster if that's what you're asking.

**Reply by ajguerre1** [reply]
Yes, as you said it would be to help handle more cameras than the TPU can handle. Is that a possible scenario?

**Reply by NickM-27** [reply]
I'm not sure, maybe just try and see if it starts?

But the coral can handle 10+ cameras (depends how many see motion at the same time) so that's quite a few to handle

**Reply by ajguerre1** [reply]
Yes, I have 30+ cameras in my instance. Currently the coral is reaching its limit and I want to see if adding CPUs might help alleviate the load. So I can start with 1 CPU detector to see if the load improves. Should I just add the CPU to the config as follows:

```
detectors:
  coral:
    type: edgetpu
    device: usb
  cpu1:
    type: cpu
    num_threads: 3
```

**Reply by NickM-27** [reply]
Might as well just try it and see what happens.

**Reply by ajguerre1** [reply]
Ok, thanks. I'll give it a try!

**Reply by Cerne17** [reply]
> Ok, thanks. I'll give it a try!

Did it work? I'm thinking about trying this

**Reply by NickM-27** [reply, marked as answer]
this is not recommended, using a cpu detector and coral will just cause every other detection to be slower leading to detection hitching.
